pkg/ffmpeg/codec_detector.go

```diff
@@ -0,0 +1,240 @@
+package ffmpeg
```

Comment:
This file would probably better named `codec_hardware.go`. I would move the hardware codec declarations into this file as well.